inbc-program/inbc/parser/source_app_parser.py

```diff
@@ -4,11 +4,19 @@
    SPDX-License-Identifier: Apache-2.0
 """
 import argparse
+import logging
 
+from ..inbc_exception import InbcException
 from ..xml_tag import create_xml_tag
 
+logger = logging.getLogger(__name__)
+
 
 def application_add(args: argparse.Namespace) -> str:
+    if (args.gpgKeyUri and args.gpgKeyName is None) or (args.gpgKeyName and args.gpgKeyUri is None):
+        raise InbcException(
+            "Source requires either both gpgKeyUri and gpgKeyName to be provided, or neither of them.")
```

Comment:
**Suggestion:** Use a more robust conditional check to handle the case where either gpgKeyUri or gpgKeyName is None. The current check could lead to a situation where one of them is an empty string, which would not raise an exception but is likely an error. [bug]
**NOTE:** Ensure that the suggested code is comprehensive. This serves as an example of the code you could use. The user should verify it after accepting the suggestion.
```suggestion
    if bool(args.gpgKeyUri) != bool(args.gpgKeyName):
        raise InbcException(
            "Source requires either both gpgKeyUri and gpgKeyName to be provided, or neither of them.")
```

Reply:
done